File: src/de/thyroff/imgsteg/utils/Channel.java
package de.thyroff.imgsteg.utils;

import java.security.InvalidParameterException;

public enum Channel {
    ALPHA(0),
    RED(1),
    GREEN(2),
    BLUE(3);

    private int channelInt;

    Channel(int channelInt) {
        this.channelInt = channelInt;
    }

    public static Channel toChannel(int c) {
        return switch (c) {
            case 0 -> ALPHA;
            case 1 -> RED;
            case 2 -> GREEN;
            case 3 -> BLUE;
            default -> null;
        };
    }

    public static Channel toChannel(boolean[] b) {
        if (b.length != 2) {
            throw new InvalidParameterException("Array has to have length 2");
        }
        if (!b[0] && !b[1]) {
            return ALPHA;
        } else if (!b[0]) {
            return RED;
        } else if (!b[1]) {
            return GREEN;
        } else {
            return BLUE;
        }
    }

    public int toInt() {
        return channelInt;
    }

    public boolean[] toBoolean() {
        boolean[] b = new boolean[2];
        switch (this) {
            case ALPHA -> {

            }
            case RED -> {
                b[1] = true;
            }
            case GREEN -> {
                b[0] = true;
            }
            case BLUE -> {
                b[0] = true;
                b[1] = true;
            }
        }
        return b;
    }
}

File: reactimgsteg/src/utils/Channel.js
const Channel = {
    ALPHA: { value: 0 },
    RED: { value: 1 },
    GREEN: { value: 2 },
    BLUE: { value: 3 },

    toChannel: function (c) {
        switch (c) {
            case 0:
                return this.ALPHA;
            case 1:
                return this.RED;
            case 2:
                return this.GREEN;
            case 3:
                return this.BLUE;
            default:
                return null;
        }
    },

    toChannelFromBoolean: function (b) {
        if (b.length !== 2) {
            throw new Error("Array has to have length 2");
        }

        if (!b[0] && !b[1]) {
            return this.ALPHA;
        } else if (!b[0]) {
            return this.RED;
        } else if (!b[1]) {
            return this.GREEN;
        } else {
            return this.BLUE;
        }
    },

    toInt: function () {
        return this.value;
    },

    toBoolean: function () {
        const b = [false, false];
        switch (this.value) {
            case 0: // ALPHA
                break;
            case 1: // RED
                b[1] = true;
                break;
            case 2: // GREEN
                b[0] = true;
                break;
            case 3: // BLUE
                b[0] = true;
                b[1] = true;
                break;
        }
        return b;
    }
};
